google-cloud-firestore/lib/google/cloud/firestore/filter.rb

```diff
@@ -0,0 +1,286 @@
+        #   require "google/cloud/firestore"
+        #
+        #   filter_1 = Google::Cloud::Firestore.Firestore.new(:population, :>=, 1000000)
+        #   filter_2 = Google::Cloud::Firestore.Firestore.new("done", "=", "false")
+        #
+        #   filter = filter_1.and(filter_2)
+        #
+        # @example Pass filter conditions in the argument
+        #   require "google/cloud/firestore"
+        #
+        #   filter_1 = Google::Cloud::Firestore.Firestore.new(:population, :>=, 1000000)
+        #
+        #   filter = filter_1.and("done", "=", "false")
+        #
+        def and *args
+          join_filters composite_filter_and, args
+        end
+
+        ##
+        # Joins filter using OR operator.
+        #
+        # @overload and(filter)
+        #
+        #   @param [::Google::Cloud::Firestore::Filter] filter
+        #
+        # @overload and(field, operator, value)
+        #
+        #    @param [FieldPath, String, Symbol] filter_or_field A field path to filter
+        #     results with.
+        #
+        #     If a {FieldPath} object is not provided then the field will be
+        #     treated as a dotted string, meaning the string represents individual
+        #     fields joined by ".". Fields containing `~`, `*`, `/`, `[`, `]`, and
+        #     `.` cannot be in a dotted string, and should provided using a
+        #     {FieldPath} object instead.
+        #    @param [String, Symbol] operator The operation to compare the field
+        #     to. Acceptable values include:
+        #
+        #     * less than: `<`, `lt`
+        #     * less than or equal: `<=`, `lte`
+        #     * greater than: `>`, `gt`
+        #     * greater than or equal: `>=`, `gte`
+        #     * equal: `=`, `==`, `eq`, `eql`, `is`
+        #     * not equal: `!=`
+        #     * in: `in`
+        #     * not in: `not-in`, `not_in`
+        #     * array contains: `array-contains`, `array_contains`
+        #    @param [Object] value A value the field is compared to.
+        #
+        # @return [Filter] New Filter object.
+        #
+        # @example Pass a Filter type object in argument
+        #   require "google/cloud/firestore"
+        #
+        #   filter_1 = Google::Cloud::Firestore.Firestore.new(:population, :>=, 1000000)
+        #   filter_2 = Google::Cloud::Firestore.Firestore.new("done", "=", "false")
+        #
+        #   filter = filter_1.or(filter_2)
+        #
+        # @example Pass filter conditions in the argument
+        #   require "google/cloud/firestore"
+        #
+        #   filter_1 = Google::Cloud::Firestore.Firestore.new(:population, :>=, 1000000)
+        #
+        #   filter = filter_1.or("done", "=", "false")
+        #
+        def or *args
+          join_filters composite_filter_or, args
+        end
+
+        ##
+        # @private
+        StructuredQuery = Google::Cloud::Firestore::V1::StructuredQuery
+
+        ##
+        # @private
+        FILTER_OPS = {
```

Comment:
I see this conversion is also done in query.rb https://github.com/googleapis/google-cloud-ruby/blob/7e6c7604888709abd91f660cb97f9c3fee593896/google-cloud-firestore/lib/google/cloud/firestore/query.rb#L1093 can we move them to a common place like convert.rb to avoid duplication ?

Reply:
Removed it because the code became redundant with these changes.